blueprints/component/index.js

```diff
@@ -0,0 +1,77 @@
+/* eslint-env node */
+// Note - this is the exact same as ember's normal component index.js
+
+var stringUtil         = require('ember-cli-string-utils');
+var pathUtil           = require('ember-cli-path-utils');
+var validComponentName = require('ember-cli-valid-component-name');
+var getPathOption      = require('ember-cli-get-component-path-option');
+var path               = require('path');
+
+var normalizeEntityName = require('ember-cli-normalize-entity-name');
+
+module.exports = {
+  description: 'Generates a component with ember-style-guide documentation in place. Name must contain a hyphen.',
+
+  availableOptions: [
+    {
+      name: 'path',
+      type: String,
+      default: 'components',
+      aliases: [
+        { 'no-path': '' }
+      ]
+    }
+  ],
+
+  fileMapTokens: function() {
+    return {
+      __path__: function(options) {
+        if (options.pod) {
+          return path.join(options.podPath, options.locals.path, options.dasherizedModuleName);
+        }
+        return 'components';
+      },
+      __templatepath__: function(options) {
+        if (options.pod) {
+          return path.join(options.podPath, options.locals.path, options.dasherizedModuleName);
+        }
+        return 'templates/components';
+      },
+      __templatename__: function(options) {
+        if (options.pod) {
+          return 'template';
+        }
+        return options.dasherizedModuleName;
+      }
+    };
+  },
+
+  normalizeEntityName: function(entityName) {
+    entityName = normalizeEntityName(entityName);
+
+    return validComponentName(entityName);
+  },
+
+  locals: function(options) {
+    var templatePath   = '';
```

Comment:
Lines 56-58 don't follow this own style guide ;-) https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#whitespace